pkg/proc/types.go

```diff
@@ -98,19 +98,14 @@ func (ctxt *loadDebugInfoMapsContext) lookupAbstractOrigin(bi *BinaryInfo, off d
 //     debug_info
 //   - After go1.11 the runtimeTypeToDIE map is used to look up the address of
 //     the type and map it directly to a DIE.
-func runtimeTypeToDIE(_type *Variable, dataAddr uint64) (typ godwarf.Type, kind int64, err error) {
+func RuntimeTypeToDIE(_type *Variable, dataAddr uint64, mds []ModuleData) (typ godwarf.Type, kind int64, err error) {
 	bi := _type.bi
 
 	_type = _type.maybeDereference()
 
 	// go 1.11 implementation: use extended attribute in debug_info
 
-	mds, err := loadModuleData(bi, _type.mem)
-	if err != nil {
-		return nil, 0, fmt.Errorf("error loading module data: %v", err)
-	}
-
-	md := findModuleDataForType(bi, mds, _type.Addr, _type.mem)
+	md := findModuleDataForType(bi, mds, _type.Addr, _type.Mem)
```

Comment:
This change is because you want to cache the module data list right?

Reply:
Yes, if module data is created every time calling findModuleDataForType, the performance will be very bad.